### PR TITLE
Enhance ghwt.py

### PR DESCRIPTION
**Pull request by alyst**
Enhance `ghwt.py`:
  * `list` command takes an optional argument: GitHub user/organization (defaults to `mesonbuild`)
  * `install` accepts "extended" wrap package names in the form `<user>/<package_name>` (the current "short" package names are also recognized, `<user>` defaults to `mesonbuild`)
  * `unpack()` prepares the proper patch zip and wrap file (the code is borrowed from `wrapweb/wrapcreator.py`)

#### Rationale
I'm converting [Ipopt](https://projects.coin-or.org/Ipopt) to meson. It depends on a number of other packages.
Currently COIN-OR projects have a [SVN "repository"](https://projects.coin-or.org/svn/BuildTools/ThirdParty) of dependencies (scripts to download the sources + custom makefiles).
The build scripts are tailored to the COIN-OR needs and therefore are sometimes very different from upstream originals.
So when I'm converting the dependencies to meson, it doesn't make sense to put them to the public WrapDB.
Extending `ghwt.py` looked like a good alternative.

### CODE REVIEW

File: ghwt.py [hunk 1]
```diff
@@ -29,80 +29,110 @@ def gh_get(url):
...
+    base_name += '-wrap'
+    zip_name = base_name + '.zip'
+    wrap_name = sproj + '.wrap'
+    zip_file = os.path.join(outdir, zip_name)
+    wrap_file = os.path.join(outdir, '..', wrap_name)
+    with zipfile.ZipFile(zip_file, 'w', compression=zipfile.ZIP_DEFLATED) as zip:
+        for root, dirs, files in os.walk(outdir):
+            for f in files:
+                abspath = os.path.join(root, f)
+                relpath = abspath[len(outdir)+1:]
```
Comment (ignatenkobrain) [inline ghwt.py]: __[Flake8]__

[E226] missing whitespace around arithmetic operator

File: ghwt.py [hunk 2]
```diff
@@ -29,80 +29,110 @@ def gh_get(url):
...
+                zip.write(abspath, os.path.join(wrap_def['directory'], relpath))
+
+    source_hash = hashlib.sha256(open(zip_file, 'rb').read()).hexdigest()
+    with open(wrap_file, 'w') as wrap:
+        url = 'file://' + os.path.abspath(zip_file)
+        wrap.write(upstream_content.rstrip())
+       	wrap.write('\n\n')
```
Comment (ignatenkobrain) [inline ghwt.py]: __[Flake8]__

[E101] indentation contains mixed spaces and tabs

Comment (ignatenkobrain) [inline ghwt.py]: __[Flake8]__

[W191] indentation contains tabs

**Comment by jpakkane** [comment]
You don't need any extra functionality for this. You can specify a wrap file that points to a git repo with all your extra changes and commits and it uses that directly. That is the recommended way to ship projects with custom changes.

`ghwt.py` is only an emergency tool to use when wrapdb is down. Its functionality is meant to be the same as wraptool proper.

**Comment by alyst** [comment]
@jpakkane Maybe I misread the documentation. Do you mean "[wrap-git]" files? I thought that those need git branches that contain both sources and meson.build overlay. Would it work if I'll use wrapdb git repo?

**Comment by jpakkane** [comment]
> Would it work if I'll use wrapdb git repo?

No. If you want to use an existing project that has Meson build definitions, you just need to create a wrap-git file for it (or use Git submodules for it if you prefer, both work).

**Pull request closed without merging**